controllers/serveController.js

```diff
@@ -1,150 +1,43 @@
-      .then(result => {
-        if (result) {
-          resolve(result.dataValues);
-        } else {
-          resolve(null);
-        }
-      })
-      .catch(error => {
-        reject(error);
-      });
-  });
-}
-
-function addGetResultsToFileRecord (fileInfo, getResult) {
-  fileInfo.fileName = getResult.file_name;
-  fileInfo.filePath = getResult.download_path;
-  fileInfo.fileType = getResult.mime_type;
-  return fileInfo;
-}
-
-function createFileRecord ({ name, claimId, outpoint, height, address, nsfw }) {
-  return {
-    name,
-    claimId,
-    outpoint,
-    height,
-    address,
-    fileName: '',
-    filePath: '',
-    fileType: '',
-    nsfw,
-  };
-}
-
-function getAssetByLongClaimId (fullClaimId, name) {
-  logger.debug('...getting asset by claim Id...');
-  return new Promise((resolve, reject) => {
-    // 1. check locally for claim
-    checkForLocalAssetByClaimId(fullClaimId, name)
-    .then(dataValues => {
-      // if a result was found, return early with the result
-      if (dataValues) {
-        logger.debug('found a local file for this name and claimId');
-        resolve(dataValues);
-        return;
-      }
-      logger.debug('no local file found for this name and claimId');
-      // 2. if no local claim, resolve and get the claim
-      db.Claim
-        .resolveClaim(name, fullClaimId)
-        .then(resolveResult => {
-          // if no result, return early (claim doesn't exist or isn't free)
-          if (!resolveResult) {
-            resolve(NO_CLAIM);
-            return;
-          }
-          logger.debug('resolve result >> ', resolveResult.dataValues);
-          let fileRecord = {};
-          // get the claim
-          lbryApi.getClaim(`${name}#${fullClaimId}`)
-          .then(getResult => {
-            logger.debug('getResult >>', getResult);
-            fileRecord = createFileRecord(resolveResult);
-            fileRecord = addGetResultsToFileRecord(fileRecord, getResult);
-            // insert a record in the File table & Update Claim table
-            return db.File.create(fileRecord);
-          })
-          .then(() => {
-            logger.debug('File record successfully updated');
-            resolve(fileRecord);
-          })
-          .catch(error => {
-            reject(error);
-          });
-        })
-        .catch(error => {
-          reject(error);
-        });
-    })
-    .catch(error => {
-      reject(error);
-    });
-  });
-}
-
-function chooseThumbnail (claimInfo, defaultThumbnail) {
-  if (!claimInfo.thumbnail || claimInfo.thumbnail.trim() === '') {
-    return defaultThumbnail;
-  }
-  return claimInfo.thumbnail;
-}
+const NO_FILE = 'NO_FILE';
 
 module.exports = {
-  getAssetByClaim (claimName, claimId) {
-    logger.debug(`getAssetByClaim(${claimName}, ${claimId})`);
+  getClaimId (channelName, channelId, name, claimId) {
+    if (channelName) {
+      return module.exports.getClaimIdByChannel(channelName, channelId, name);
```

Comment:
`channelId` is still a claim id. I might call this something like `channelClaimId` to make that clearer.